llvm/lib/Target/SystemZ/SystemZISelDAGToDAG.cpp

```diff
@@ -1545,6 +1552,37 @@ bool SystemZDAGToDAGISel::storeLoadIsAligned(SDNode *N) const {
   return true;
 }
 
+// This is a hack to convert ATOMIC_LOADs to LOADs in the last minute just
```

Comment:
It's also not difficult to use PatFrags between atomic and non-atomic loads instead of hacking this up in C++

Reply:
Well, with all the different extending load operators it is not quite simple I'd say, but of course better. My concern is that if the ATOMIC_LOAD opcode will go away those changes would have to be reverted again in the .td files. So if that might happen in some foreseeable future, maybe this hack would be acceptable? As an alternative, maybe the switch to PatFrags could be done as a follow-up patch so that it could more easily be reverted if needed..?

Reply:
I still think getting rid of ATOMIC_LOAD is the correct way to do this, and anything else is working around an aversion to completing the work to get there